crates/wasmparser/src/validator/types.rs

```diff
@@ -1279,12 +1256,13 @@ impl<'a> TypesRef<'a> {
 
     /// Gets the type of an component instance at the given component instance index.
     ///
-    /// Returns `None` if the type index is out of bounds or the type has not
-    /// been parsed yet.
-    pub fn component_instance_at(&self, index: u32) -> Option<TypeId> {
+    /// # Panics
+    ///
+    /// This will panic if the `index` provided is out of bounds.
```

Comment:
This function and several other similar functions also panic if `self` is a core module.